games/loveletter/rules.js

```diff
@@ -0,0 +1,371 @@
+  /* eslint-enable */
+}
+
+function is_handmaided(player_discards) {
+  return (
+    (player_discards.length > 0)
+    && player_discards[player_discards.length - 1] === 4
+  );
+}
+
+function num_wins_required(num_players) {
+  if (num_players === 2) return 7;
+  if (num_players === 3) return 5;
+  if (num_players === 4) return 4;
+  if (num_players === 5) return 3;
+  throw 'Invalid number of players in num_wins_required';
+}
+
+function new_round(cur_game_state) {
+  const game_state = cur_game_state;
+  const { num_players } = game_state;
+  game_state.deck = make_deck();
+  game_state.hands = new Array(num_players);
+  for (let i = 0; i < num_players; i += 1) {
+    game_state.hands[i] = [];
+    pickup_card(game_state, i);
+  }
+  game_state.discards = new Array(num_players).fill([]);
+  game_state.priested_player = null;
+  return game_state;
+}
+
+function initial_state(players) {
+  if (players < 2 || players > 4) {
+    throw 'Invalid number of players';
+  }
+
+  const current_player = 0; // TODO: different current player?
+  const game_state = {
+    log: ['Game started.'],
+    num_wins: new Array(players).fill(0), // Number of wins for each player
+    num_players: players,
+    current_player,
+  };
+  new_round(game_state, current_player);
+  pickup_card(game_state, game_state.current_player);
+  return game_state;
+}
+
+function player_view(game_state, player_id) {
+  return {
+    player_id,
+    log: game_state.log,
+    cards_left: game_state.deck.length,
+    hands: game_state.hands.map((h, i) => h.map(c => (
+      (i === player_id)
+      || (game_state.priested_player === i && game_state.current_player === player_id)
+      ? c : null
+    ))), // Hide card information from player
+    discards: game_state.discards,
+    player_alive: game_state.hands.map(h => h.length > 0),
+    num_wins: game_state.num_wins,
+    num_players: game_state.num_players,
+    priested_player: game_state.priested_player,
+    current_player: game_state.current_player,
+  };
+}
+
+function current_players(game_state) {
+  return [game_state.current_player];
+}
+
+function has_legal_action(game_view) {
+  // NOT ACTUALLY USED LOL
+  return game_view.player_id === game_view.current_player;
+}
+
+function is_action_legal(game_view, action) {
+  let all_others_handmaided = true;
+  for (let i = 0; i < game_view.num_players; i += 1) {
+    if (i !== game_view.current_player && is_handmaided(game_view.discards[i])) {
+      all_others_handmaided = false;
+      break;
+    }
+  }
+
+  // Player must be current player
+  if (game_view.player_id !== game_view.current_player) return false;
+  // If the player just played a priest, they're pressing "continue" and their
+  // action will be rewritten to continue the game. All other bets are off.
+  if (game_view.priested_player != null) return true;
+  // Player must have selected card
+  if (!game_view.hands[game_view.player_id].includes(action.card)) return false;
+  if (action.target) {
```

Comment:
action.target !== undefined. Currently the game always lets you target player 0 because none of the checks run.

Reply:
Fixed.